invenio_rdm_records/services/access/service.py

```diff
@@ -506,97 +531,21 @@ def read_all_grants(self, identity, id_, expand=False):
         # Permissions
         self.require_permission(identity, "manage", record=record)
 
-        # Fetching
-        return self.grants_result_list(
-            service=self,
-            identity=identity,
-            results=parent.access.grants,
-            expand=expand,
-        )
+        existing_grants = parent.access.grants
 
-    def read_all_grants_by_subject(self, identity, id_, subject_type, expand=False):
-        """Read access grants of a record (resp. its parent) by subject type."""
-        record, parent = self.get_parent_and_record_or_draft(id_)
-
-        # Permissions
-        self.require_permission(identity, "manage", record=record)
-
-        user_grants = []
-        for grant in parent.access.grants:
-            if grant.subject_type == subject_type:
-                user_grants.append(grant)
+        for grant in existing_grants:
+            # removes group grants if groups are not enabled in the instance
+            if (
+                not current_app.config.get("USERS_RESOURCES_GROUPS_ENABLED", False)
+                and grant.subject_type == RecordAccessService.group_subject_type
+            ):
+                existing_grants.remove(grant)
```

Comment:
Note, for `read_all` we don't fail with 403, instead we return only user grants, even if group grants are present.

Reply:
could you add the note in the code as a comment?